app/databaseAccessors/articleAccessor.js

```diff
@@ -388,36 +388,25 @@ export default class ArticlesAccessor {
   }
 
   /**
-   * addCommentBySlug method
+   * resolveCommentById method
    *
-   * This method finds the article with the given
-   * slug and adds the given comment to its comments.
+   * This method finds the comment with the ID
+   * and resolves it.
    *
-   * @param {slug}  slug of the article to find
-   * @param {comment} comment to be added to the article
-   * @returns a single updated article
+   * @param {commentId} mongo Id of the comment to be resolved
    */
-  static async addCommentBySlug(slug, comment) {
+  static async resolveCommentById(commentId) {
     try {
       await Connection.open();
-      // update the article by adding the new comment to its array
-      const newArticle = await Article.findOneAndUpdate(
-        { slug: slug },
+      // find the comment by ID and then resolve it
+      await Article.findOneAndUpdate(
+        { "comments._id": commentId },
         {
-          $push: {
-            comments: comment,
+          $set: {
+            "comments.$.commentStatus": "resolved",
```

Comment:
Interesting query, I haven't seen this before. Do you mind sharing the docs for where you found how to construct a query like this for reference?